packages/queued-request-controller/src/QueuedRequestMiddleware.ts

```diff
@@ -53,124 +23,69 @@ const isConfirmationMethod = (method: string) => {
-            networkConfigurationForRequest =
-              networkControllerState.networkConfigurations[
-                networkClientIdForRequest
-              ];
-          } else {
-            // if its a built in
-            // Ideally we should be using only networkConfigurations, and networkClientIds &
-            // networkConfiguration.id should be the same thing.
-            networkConfigurationForRequest =
-              networkClientConfigurationForRequest;
-          }
-
-          const currentProviderConfig = networkControllerState.providerConfig;
-          const currentChainId = currentProviderConfig.chainId;
-
-          // if the 'globally selected network' is already on the correct chain for the request currently being processed
-          // continue with the request as normal.
-          if (currentChainId === networkConfigurationForRequest.chainId) {
-            return next();
-          }
-
-          // todo once we have 'batches':
-          // if is switch eth chain call
-          // clear request queue when the switch ethereum chain call completes (success, but maybe not if it fails?)
-          // This is because a dapp-requested switch ethereum chain invalidates any requests they've made after this switch, since we dont know if they were expecting the chain after the switch or before.
-          // with the queue batching approach, this would mean clearing any batch for that origin (batches being per-origin.)
-          const requestData = {
-            toNetworkConfiguration: networkConfigurationForRequest,
-            fromNetworkConfiguration: currentProviderConfig,
-          };
-
-          try {
-            await messenger.call(
-              'ApprovalController:addRequest',
-              {
-                origin,
-                type: ApprovalType.SwitchEthereumChain,
-                requestData,
-              },
-              true,
-            );
-
-            await messenger.call(
-              `NetworkController:setActiveNetwork`,
-              networkClientIdForRequest,
-            );
-
-            messenger.call(
-              SelectedNetworkControllerActionTypes.setNetworkClientIdForDomain,
-              origin,
-              networkClientIdForRequest,
-            );
-          } catch (error) {
-            res.error = serializeError(error);
-            return error;
-          }
-
-          return next();
-        },
-      );
-    },
-  );
+  return createAsyncMiddleware(async (req: JsonRpcRequest, res, next) => {
+    hasRequiredMetadata(req);
+
+    // if the request queue feature is turned off, or this method is not a confirmation method
+    // bypass the queue completely
+    if (!useRequestQueue() || !isConfirmationMethod(req.method)) {
+      return await next();
+    }
+
+    try {
+      await enqueueRequest(req, async (error: unknown) => {
+        if (error) {
+          res.error = serializeError(error);
+          return undefined;
+        }
+
```

Comment:
It seems that `enqueueRequest` takes a function but never calls it with any arguments, so can this be simplified?

``` suggestion
      await enqueueRequest(req, async () => {
```

Reply:
Looks like `requestNext` is invoked with an `error` argument in the `QueuedRequestMiddleware.test.ts` file `getMockEnqueueRequest` function, and the function is called with `Error` constructors.

Reply:
Simplified in 80f305c29084a9bead31942cbd3122fcd34409a4